.github/actions/download-plugin/action.yml

```diff
@@ -17,20 +17,12 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Download OpenSearch for Linux
-      uses: peternied/download-file@v2
-      if: ${{ runner.os == 'Linux' }}
-      with:
-        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/plugins/${{ inputs.plugin-name }}-${{ inputs.plugin-version }}.zip
-
-    - name: Download OpenSearch for Windows
-      uses: peternied/download-file@v2
-      if: ${{ runner.os == 'Windows' }}
-      with:
-        url: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/windows/x64/zip/builds/opensearch/plugins/${{ inputs.plugin-name }}-${{ inputs.plugin-version }}.zip
-
-    - name: Rename the Plugin Files
-      run: mv opensearch-security-${{ inputs.plugin-version }}.zip opensearch-security.zip
+    - run: |
+        mvn dependency:get \
+        -DremoteRepositories=https://aws.oss.sonatype.org/content/repositories/snapshots/ \
+        -Dartifact=org.opensearch.plugin:${{ inputs.plugin-name }}:${{ inputs.plugin-version }}-SNAPSHOT:zip \
```

Comment:
`-SNAPSHOT` should be removed here and should be passed dynamically in the `plugin-version` variable.

Reply:
This repository only storage snapshots, `-D remoteRepositories=https://aws.oss.sonatype.org/content/repositories/snapshots/`